Draft-Accepted/RFCXXXX-FeedbackProviders.md

```diff
@@ -0,0 +1,192 @@
+---
+RFC: RFC<four digit unique incrementing number assigned by Committee, this shall be left blank by the author>
+Author: Steven Bucher
+Status: Draft
+SupercededBy: NA
+Version: 1.0
+Area: Interactive Shell
+Comments Due: July 1st, 2023
+Plan to implement: Yes
+---
+
+# Feedback Providers
+
+Feedback Providers are PowerShell modules that utilize the `IFeedbackProvider` interface to trigger
+feedback on errors, comments, and success events. These providers can help enhance user experience
+by providing real-time feedback, notifications, or suggestions based on the execution of PowerShell
+commands. This RFC outlines some of the requirements and scenarios for Feedback Providers
+
+## Motivation
+
+    As a PowerShell interactive user,
+    I can get feedback on the commands I am running and attemping to run,
+    so that I can successfully use the CLI quicker and without distruption to my workflow.
+
+## Specification
+_____
+
+## User Interface and User Experience
+
+Feedback providers are triggered on the following:
+- **Comments**
+- **Success**
+- **CommandNotFound**
+- **Error**
+- **All**
+
+The current user interface schema is as follows:
+```
+[Name of first feedback provider]
+  Required Header:
+    ➤ optional action 1
+    ➤ optional action 2
+    ➤ optional action n
+  optional Footer
+...
+[Name of n feedback provider]
+  Required Header:
+    ➤ optional action 1
+    ➤ optional action 2
+    ➤ optional action n
+  optional Footer
+```
+
+> Note: For the sake of collaboration I am including feedback provider UIs in text rather than
+> screenshots. This means the coloring will be off to the final experience.
+
+### On Comment
+
+Comments allow for a more descriptive input for feedback providers to utilize and act upon. One
+example implementation could be using a feedback provider as a pass through to an off box A.I model.
+This way the more detailed input can be used by a large language model.
+
+```powershell
+PS># How do I list the files in my C drive?
+[A.I Model]
+  One way to list all your files in the C drive is the following:
+    ➤ Get-ChildItem -Path C:
+```
+
+### On Success
```

Comment:
We should break down success cases so that FeedbackProviders can be more specialized and don't get invoked for every case.   For example, cmdlet vs native command vs script